src/Input/Keybindings.re

```diff
@@ -84,9 +91,49 @@ module Internal = {
     };
 };
 
-let empty = [];
+module Input =
+  EditorInput.Make({
+    type context = Hashtbl.t(string, bool);
+    type payload = string;
+  });
+
+type effect =
+  | Command(string)
+  | Text(string)
+  | Unhandled(EditorInput.keyPress);
+
+let empty = Input.empty;
+
+type t = Input.t;
+
+let mapEffect =
+  fun
+  | Input.Execute(cmd) => Command(cmd)
+  | Input.Unhandled(key) => Unhandled(key)
+  | Input.Text(text) => Text(text);
+
+let mapEffects = List.map(mapEffect);
+
+let keyDown = (~context, ~key, bindings) => {
+  let (bindings, effects) = Input.keyDown(~context, ~key, bindings);
+
+  let mappedEffects = mapEffects(effects);
```

Comment:
`mapEffect`, `mapEffects` and `mappedEffects` is very non-descriptive. The most significant information, what it maps from and to, is missing.

Reply:
I agree, but I don't have ideas for better names (except for something overly verbose like `mapEditorInputEffectToOnivim2Effect`, and I'm not sure that's great). 

Since it's an internal implementation detail, and there isn't a better name suggested, I'll leave it - but I moved it to the `Internal` module and tried to describe it via a comment.

Reply:
It might be hard to name because it really doesn't seem to do anything "useful", other than perhaps avoid some technical limitation.

I wonder why the local `effect` type isn't just an alias of the one from `Input` though, as they seem to have identical definitions. That would avoid the need to map it at all.

Reply:
Good point. The challenge was that the `effect` type was tucked inside the functorized-`EditorInput.Make`. I extracted it out so that it could be aliased from `EditorInput`, which removed these extra maps 👍